build.sh

```diff
@@ -112,7 +112,7 @@ prepare_native_build()
     # Generate version.c if specified, else have an empty one.
     __versionSourceFile=$__scriptpath/bin/obj/version.c
     if [ $__generateversionsource == true ]; then
-        $__scriptpath/Tools/corerun $__scriptpath/Tools/MSBuild.exe "$__scriptpath/build.proj" /t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__scriptpath/bin/obj/version.c /p:GenerateVersionSourceFile=true /v:minimal
```

Comment:
When I'm reading the scripts having arguments helps me to understand what is happening. This arguments by themselves are not clear and if we need them all together in order to work I prefer to have them encapsulated in a property.

What we've been trying to do is that if we need to pass just a property to a script in order to activate something then there is no need to create an argument for the script, we will just have documentation about it. But if we need a set of properties/targets/etc (like this case) create an argument is preferred. Is that correct @weshaggard ?

Currently I'm working on a common setup script (cmd/sh) that will encapsulate all the properties we need across the dev workflow scripts, so if we found that a "build official" property needs to be set in other scripts, we could defined it in the common setup script.